### PR TITLE
fix a java course link by programmerzamannow

### PR DESCRIPTION
**Pull request by nurulramadhona**
## What does this PR do?
Add resource(s) | Remove resource(s)

### Description
The existing link is no longer available. So, I updated with the new one.
### How do we know it's really free?
You can directly access it.
### For book lists, is it a book? For course lists, is it a course? etc.
It's a course
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-id.md [hunk 1]
```diff
@@ -127,7 +127,7 @@
 * [Belajar Java - Object Oriented Programming](https://www.youtube.com/playlist?list=PLFfUPa9IV8LpbRH5-TzphcZj6tpoxdr-p) - Mastahcode
 * [Belajar Java - Object Oriented Programming Bahasa Indonesia (Lanjut)](https://www.youtube.com/playlist?list=PLZS-MHyEIRo6V4_vk1s1NcM2HoW5KFG7i) - Kelas Terbuka
 * [Belajar Java Bahasa Indonesia (Dasar)](https://www.youtube.com/playlist?list=PLZS-MHyEIRo51w0Hmqi0C8h2KWNzDfo6F) - Kelas Terbuka
-* [Belajar Java Lambda Expression](https://www.youtube.com/playlist?list=PL-CtdCApEFH_71iT2klDSWIC0MScpLheN) - Programmer Zaman Now
+* [JAVA TUTORIAL - BAHASA INDONESIA](https://www.youtube.com/playlist?list=PL-CtdCApEFH-p_Q2GyK4K3ORoAT0Yt7CX) - Programmer Zaman Now
```
Comment (davorpa) [inline courses/free-courses-id.md:130]: - Although channel is named `Programmer Zaman Now` the creator is @khannedy
- We also take the opportunity to complete with the resource platform

```markdown
* [JAVA TUTORIAL - BAHASA INDONESIA](https://www.youtube.com/playlist?list=PL-CtdCApEFH-p_Q2GyK4K3ORoAT0Yt7CX) - Eko Kurniawan Khannedy, Programmer Zaman Now (YouTube)
```

**Review by davorpa — CHANGES_REQUESTED** [review]
Nice catch.

Helps to improve:
- #5470

**Comment by nurulramadhona** [comment]
I've updated the creator's name. Not only on the resource I fixed the link, but all resources that use the same creator.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**